### PR TITLE
[Tests-Only]Adjust only docs changes

### PR DESCRIPTION
**Pull request by jasson99**
## Description
This PR adjusts the `.drone.star` file and `getFilesChanged.sh` (previously `if-diff-has-unit-tests-only.sh`) file such that:
- when the changes are only in `docs-only` file, then other pipelines are skipped
- when the changes are in `docs` and `unit-tests` then the unit tests pipelines are run

## Related Issue
- https://github.com/owncloud/web/issues/5637#issuecomment-894025552

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalGroupsToRoot</strong> https://drone.owncloud.com/owncloud/web/18577/23/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Resharing</strong> https://drone.owncloud.com/owncloud/web/18577/15/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPermissionToRoot</strong> https://drone.owncloud.com/owncloud/web/18577/32/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalGroups</strong> https://drone.owncloud.com/owncloud/web/18577/22/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersRoot1</strong> https://drone.owncloud.com/owncloud/web/18577/29/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingFolderPermissions</strong> https://drone.owncloud.com/owncloud/web/18577/21/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublicBasic</strong> https://drone.owncloud.com/owncloud/web/18577/33/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: tests/drone/getFilesChanged.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	 then
+		 CHANGED_UNIT_TESTS_AND_DOCS_ONLY=False
+		 break
+	 fi
+	fi
+done
+
+if [ ! "${DIFF}" ]
+then
+	echo "no any files are changed"
```
Comment (pascalwengerter) [inline tests/drone/getFilesChanged.sh:33]:
```suggestion
	echo "no files are changed"
```

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/18589/65/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/18589/13/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Basic</strong> https://drone.owncloud.com/owncloud/web/18589/7/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone2</strong> https://drone.owncloud.com/owncloud/web/18589/46/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationNotifications</strong> https://drone.owncloud.com/owncloud/web/18589/64/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersBlacklisted</strong> https://drone.owncloud.com/owncloud/web/18589/26/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Rename</strong> https://drone.owncloud.com/owncloud/web/18589/10/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Locks</strong> https://drone.owncloud.com/owncloud/web/18589/8/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalGroupsSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18589/24/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Sharing1</strong> https://drone.owncloud.com/owncloud/web/18589/16/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublicExpireAndRoles</strong> https://drone.owncloud.com/owncloud/web/18589/35/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Sharing1</strong> https://drone.owncloud.com/owncloud/web/18592/16/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone2</strong> https://drone.owncloud.com/owncloud/web/18592/46/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublicBasic</strong> https://drone.owncloud.com/owncloud/web/18592/33/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10MoveUpload</strong> https://drone.owncloud.com/owncloud/web/18592/14/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhoneNotifications</strong> https://drone.owncloud.com/owncloud/web/18592/44/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/18592/65/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingFilePermission</strong> https://drone.owncloud.com/owncloud/web/18592/18/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait2</strong> https://drone.owncloud.com/owncloud/web/18592/43/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Basic</strong> https://drone.owncloud.com/owncloud/web/18592/7/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingNotifications</strong> https://drone.owncloud.com/owncloud/web/18592/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternal</strong> https://drone.owncloud.com/owncloud/web/18592/39/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone1</strong> https://drone.owncloud.com/owncloud/web/18592/45/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsers</strong> https://drone.owncloud.com/owncloud/web/18592/25/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersRoot1</strong> https://drone.owncloud.com/owncloud/web/18592/29/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationNotifications</strong> https://drone.owncloud.com/owncloud/web/18592/64/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait1</strong> https://drone.owncloud.com/owncloud/web/18592/42/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18592/28/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortraitNotifications</strong> https://drone.owncloud.com/owncloud/web/18592/41/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersRoot2</strong> https://drone.owncloud.com/owncloud/web/18592/30/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPermissionsUsers</strong> https://drone.owncloud.com/owncloud/web/18595/31/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPermissionToRoot</strong> https://drone.owncloud.com/owncloud/web/18595/32/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingFolderAdvPermissionsGrp</strong> https://drone.owncloud.com/owncloud/web/18595/20/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Rename</strong> https://drone.owncloud.com/owncloud/web/18595/10/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/18595/13/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersBlacklisted</strong> https://drone.owncloud.com/owncloud/web/18595/26/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublicManagement</strong> https://drone.owncloud.com/owncloud/web/18595/34/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalGroupsToRoot</strong> https://drone.owncloud.com/owncloud/web/18595/23/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: .drone.star [hunk 2]
```diff
@@ -2276,7 +2277,7 @@ def listRemoteCache():
             },
         },
         "commands": [
-            "if test -f runUnitTestsOnly; then echo 'skipping list-ocis-bin-cache'; else mc alias set s3 $MC_HOST $AWS_ACCESS_KEY_ID $AWS_SECRET_ACCESS_KEY && mc find s3/owncloud/web/ocis-build; fi",
+            "if test -f runUnitTestsOnly || test -f runTestsForDocsChangeOnly; then echo 'skipping list-ocis-bin-cache'; else mc alias set s3 $MC_HOST $AWS_ACCESS_KEY_ID $AWS_SECRET_ACCESS_KEY && mc find s3/owncloud/web/ocis-build; fi",
```
Comment (kiranparajuli589) [inline .drone.star:2280]:
```suggestion
            "if test -f doNotRunAcceptanceTests; then echo 'skipping list-ocis-bin-cache'; else mc alias set s3 $MC_HOST $AWS_ACCESS_KEY_ID $AWS_SECRET_ACCESS_KEY && mc find s3/owncloud/web/ocis-build; fi",
```
just a suggestion 🤔  if you can create a file `doNotRunAcceptanceTests` too for the cases where we need to skip acceptance suites, then we do not have to check for two files.

File: tests/drone/getFilesChanged.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+CHANGED_UNIT_TESTS_ONLY=True
+CHANGED_DOCS_ONLY=True
+CHANGED_UNIT_TESTS_AND_DOCS_ONLY=True
+
+DIFFS=$(git diff origin/master --name-only)
+DOCS_ONLY_CHANGES=$(echo "$DIFFS" | grep -E '(docs/.*|changelog/.*)')
+UNIT_TESTS_CHANGES=$(echo "$DIFFS" | grep 'packages/.*/tests/.*')
+DOCS_AND_UNIT_TESTS_ONLY_CHANGES=$(echo "$DIFFS" | grep -E '(docs/.*|changelog/.*|packages/.*/tests/.*)')
```
Comment (dpakach) [inline tests/drone/getFilesChanged.sh:10]: will these regex hold if the path falls inside the docs folder?
for eg. if a file is changed with path `docs/.../packages/files/tests/unitTest.md` then you will get this in both docs only changes and unit tests changes but in reality only docs has been changed.

File: tests/drone/getFilesChanged.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+CHANGED_UNIT_TESTS_ONLY=True
+CHANGED_DOCS_ONLY=True
+CHANGED_UNIT_TESTS_AND_DOCS_ONLY=True
+
+DIFFS=$(git diff origin/master --name-only)
+DOCS_ONLY_CHANGES=$(echo "$DIFFS" | grep -E '(docs/.*|changelog/.*)')
```
Comment (dpakach) [inline tests/drone/getFilesChanged.sh:8]: I think this name is a bit confusing to me. This gives weather there is docs related changes (there can be other changes too), but the var name is `DOCS_ONLY_CHANGES` which (only part)  is not really true.

**Review by dpakach — COMMENTED** [review]
Just a general suggestion for the bash script, rather than trying to find out if there has been docs only changes or unit tests only changes, I would first evaluate 3 things 
a. is there unit tests changes 
b. is there docs changes
c. is there any other type of changes (source, deps, config, accpetance tests etc.)
These can be evaluated fairly easily with regex. Also use regex patterns to match the path starting the beginning from each line so you dont get issues as I mentioned in my review. 
After evaluating those 3 variables (true/false) you can just use couple of if-else blocks to do everything.

**Review by kulmann — COMMENTED** [review]
Instead of introducing different boolean variables, can't we just have one bitmask that represents (per bit), if unit tests should run, integration tests should run, docs should be built, changelog should be built, e2e tests should run etc? and then use that bitmask for checking if a certain pipeline should run or not? I don't know too much about the drone config, so just think of this as a naive question please. Still a valid one IMO. 😁

File: tests/drone/getFilesChanged.sh (hunk 4 above)
Comment (jasson99) [reply tests/drone/getFilesChanged.sh]: only the changes in the files related to docs are extracted hence the name `docs_only_changes`. these changes have been used below to identify other conditions

File: .drone.star (hunk 2 above)
Comment (jasson99) [reply .drone.star:2280]: `doNotRunAcceptanceTests` would then be needed for `docs only` changes, `unit tests only` changes, and for `combination of docs and unit tests only` changes. But we can to run only docs related pipelines in `docs only` changes, skipping `unit tests` as well, but in `unit tests` only or a combination of `docs and unit tests only`, we want to run `unit tests` as well.

File: tests/drone/getFilesChanged.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+CHANGED_UNIT_TESTS_ONLY=True
+CHANGED_DOCS_ONLY=True
+CHANGED_UNIT_TESTS_AND_DOCS_ONLY = True
```
[the remark this answers is not in the document]
Comment (jasson99) [reply tests/drone/getFilesChanged.sh]: There has been some changes now, but the values are set true first. Please review again.

File: tests/drone/getFilesChanged.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	fi
+	if ! echo "${DIFF}" | grep -E '(docs/.*|changelog/.*)'
+	then
+		CHANGED_DOCS_ONLY=False
+		break
+	fi
+	if [ $CHANGED_UNIT_TESTS_ONLY == "False" && $CHANGED_DOCS_ONLY == "False" ]
+	then
+	 if ! echo "${DIFF}" | grep -E '(docs/.*|changelog/.*|packages/.*/tests/.*)'
+	 then
```
[the remark this answers is not in the document]
Comment (jasson99) [reply tests/drone/getFilesChanged.sh]: the logics have been altered a bit now

**Comment by kulmann** [comment]
What's the state of this PR?

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingPublicManagement</strong> https://drone.owncloud.com/owncloud/web/19155/34/1
The following scenarios passed on retry:
- webUISharingPublicManagement/publicLinkIndicator.feature:27

**Comment by ownclouders** [comment]
Results for <strong>oCISRename</strong> https://drone.owncloud.com/owncloud/web/19155/49/1
The following scenarios passed on retry:
- webUIRenameFolders/renameFolders.feature:107

File: .drone.star (hunk 2 above)
Comment (phil-davis) [reply .drone.star:2280]: There is some refactoring that could be done here. But lets get this merged, and check that it works OK. Then we can adjust as needed.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
I squashed and rebased. It should run all CI pipelines, because there are changes to `.drone.star` and acceptance test scripts etc. After this is merged, I will try some PRs that change small combinations of docs and unit test files...

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersRoot2</strong> https://drone.owncloud.com/owncloud/web/19166/30/1
The following scenarios passed on retry:
- webUISharingInternalUsersToRootShareWithPage/shareWithUsers.feature:79

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/web/19166/6/5 - unit tests ran
https://drone.owncloud.com/owncloud/web/19166/6/6 - integration tests ran
https://drone.owncloud.com/owncloud/web/19166/7/15 - UI test pipelines ran

Looks good.

**Pull request merged**